### PR TITLE
Add gulp

### PR DESCRIPTION
**Pull request by sannykr**

### CODE REVIEW

File: recipes/gulp/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+{% set version = "3.9.1" %}
+{% set sha256 = "9616c6bc8ca165375832bdc096260931369d5339edef2e0a4d48bc7a05b04d15" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://github.com/gulpjs/gulp/archive/v3.9.1.tar.gz
```
Comment (dougalsutherland) [inline recipes/gulp/meta.yaml]: `v{{ version }}.tar.gz`

File: recipes/gulp/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+build:
+  number: 0
+  noarch: True
+  script: npm install -g gulp
+
+requirements:
+  build:
+    - nodejs
+  run:
```
Comment (dougalsutherland) [inline recipes/gulp/meta.yaml]: Needs `nodejs` at runtime too.

**Comment by dougalsutherland** [comment]
The test failure now is because gulp, and the npm ecosystem in general, really aren't designed to be used this way. See e.g. https://stackoverflow.com/q/22115400/344821.

**Comment by jakirkham** [comment]
Maybe post-link and pre-unlink scripts would help with that.

**Comment by dougalsutherland** [comment]
I don't think so; `npm link gulp` is for the current project directory, not some general step that only needs to be done once.

**Comment by sannykr** [comment]
@conda-forge/staged-recipes : Could you please review and merge.

File: recipes/gulp/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  number: 0
+  noarch: generic
+  script: 
+    - npm install -g gulp
+ 
+requirements:
+  build:
+    - nodejs
+  run:
+    - nodejs
```
Comment (isuruf) [inline recipes/gulp/meta.yaml]: Can you please fix this as in the rest of the PRs?

Comment (dougalsutherland) [reply recipes/gulp/meta.yaml]: This one actually makes sense I think; `gulp` is a command-line build tool, and isn't useful without `node`.

File: recipes/gulp/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+       
+test:
+  commands:
```
Comment (dougalsutherland) [inline recipes/gulp/meta.yaml]: I'd be happier if this had a test that the script actually worked. Something like `npm link gulp && gulp --help` might work.

Comment (sannykr) [reply recipes/gulp/meta.yaml]: Thanks for the review. I have added this the test/commands

**Comment by dougalsutherland** [comment]
Oops, there isn't actually a `gulp --help`, because who needs thirty years of established command line conventions when you have javascript? I added a test that actually works.

**Comment by sannykr** [comment]
Looks like it's good to go. Could you please review and let me know if you have more suggestions.

**Comment by dougalsutherland** [comment]
Okay, merged. Thanks @sannykr!

**Pull request merged**